src/lightning/fabric/strategies/launchers/subprocess_script.py

```diff
@@ -154,6 +155,8 @@ def _hydra_subprocess_cmd(local_rank: int) -> Tuple[Sequence[str], str]:
     command += sys.argv[1:]
 
     cwd = get_original_cwd()
-    os_cwd = f'"{os.getcwd()}"'
-    command += [f"hydra.run.dir={os_cwd}", f"hydra.job.name=train_ddp_process_{local_rank}"]
+    run_dir = f'"{HydraConfig.get().run.dir}"'
+    command += [f"hydra.run.dir={run_dir}", f"hydra.job.name=train_ddp_process_{local_rank}"]
+    subdir = "null" if HydraConfig.get().output_subdir is None else f".pl_ddp_hydra_{local_rank}"
```

Comment:
The `.pl_ddp_hydra_local_rank` is a new addition here too. This is the subdir we choose if the hydra config does not provide the output dir. Is this the default behavior, correct? Perhaps we should add an additional assertion to the test that this folder got created.

Reply:
`.pl_ddp_hydra_{local_rank}` is actually a subdir, not the output dir (basically, a replacement to the `.hydra` subdirectory that's created inside the output dir).

I think this change makes sense because otherwise all the different processes will try to write the `config.yaml` file inside the same `.hydra` folder, which might lead to the file being garbled (not sure if the writes by different processes are atomic).

Reply:
Only hydra writes to these special folders, right? The user wouldn't see or interact with files in there (an immense noob question sorry).

Reply:
The purpose of the special folders is to dump the configuration used for the job, if the user wants to access them later.

Reply:
Okay I see. To me that sounds a bit strange, I can' see why the configuration file should be different between the processes. So the user would end up with many folders (e.g. 8 folders per machine if 8 gpus are used and all have the same config file inside). That might be a bit confusing. 

My proposal:
For (global) rank > 0, set `output_subdir` to null (disabling the saving of the config files). Then, only rank 0 will write the `.hydra` folder.

Reply:
Makes sense. Done.